std/haxe/ds/BalancedTree.hx

```diff
@@ -128,6 +128,12 @@ class BalancedTree<K,V> implements haxe.Constraints.IMap<K,V> {
 		keysLoop(root, ret);
 		return ret.iterator();
 	}
+	
+	public function copy():BalancedTree<K, V> {
+		var copied = new BalancedTree<K, V>();
+		for(key in keys()) copied.set(key, get(key));
```

Comment:
That reminds me I wanted to suggest exposing some function to save and restore the map state, but I couldn't think of a good API.